codex-rs/core/src/client_common.rs

```diff
@@ -38,12 +38,41 @@ pub struct Prompt {
 }
 
 impl Prompt {
-    pub(crate) fn get_full_instructions(&self, model: &str) -> Cow<'_, str> {
-        let base = self
+    pub(crate) fn get_full_instructions(
+        &self,
+        model: &str,
+        include_plan_tool: bool,
```

Comment:
Can we introduce `struct InstructionsConfig { include_plan_tool: bool }` and pass that in so it's more self-documenting?

We could probably eliminate `model` as a param and make `extra_apply_patch_tool_instructions` or something a field of `InstructionsConfig` and then do `InstructionsConfig::new(model: &str)` so it sets `extra_apply_patch_tool_instructions` based on `model`?